src/test/java/io/gravitee/policy/dynamicrouting/DynamicRoutingPolicyIntegrationTest.java

```diff
@@ -0,0 +1,136 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.dynamicrouting;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import io.gravitee.apim.gateway.tests.sdk.AbstractPolicyTest;
+import io.gravitee.apim.gateway.tests.sdk.annotations.DeployApi;
+import io.gravitee.apim.gateway.tests.sdk.annotations.GatewayTest;
+import io.gravitee.apim.gateway.tests.sdk.configuration.GatewayConfigurationBuilder;
+import io.gravitee.definition.model.Api;
+import io.gravitee.definition.model.ExecutionMode;
+import io.gravitee.gateway.reactor.ReactableApi;
+import io.gravitee.policy.dynamicrouting.configuration.DynamicRoutingPolicyConfiguration;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.rxjava3.core.http.HttpClient;
+import io.vertx.rxjava3.core.http.HttpClientRequest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@GatewayTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DynamicRoutingPolicyIntegrationTest extends AbstractPolicyTest<DynamicRoutingPolicy, DynamicRoutingPolicyConfiguration> {
+
+    private final WireMockServer group1WireMock = new WireMockServer(wireMockConfig().dynamicPort());
+
+    @Override
+    protected void configureGateway(GatewayConfigurationBuilder gatewayConfigurationBuilder) {
+        super.configureGateway(gatewayConfigurationBuilder);
+        gatewayConfigurationBuilder.set("api.jupiterMode.enabled", "true");
+    }
+
+    @Override
+    public void configureApi(Api api) {
+        super.configureApi(api);
+        api.setExecutionMode(ExecutionMode.JUPITER);
+    }
+
+    @BeforeAll
+    void beforeAll() {
+        group1WireMock.start();
+    }
+
+    @AfterAll
+    void afterAll() {
+        group1WireMock.stop();
+    }
+
+    @Test
+    @DeployApi(value = { "/apis/v2/with_endpoint.json" })
+    public void should_route_to_another_group_when_request_match(HttpClient client) throws InterruptedException {
+        group1WireMock.stubFor(get("/my-store").willReturn(ok()));
+
+        client
+            .rxRequest(HttpMethod.GET, "/with_endpoint/v1/stores/my-store")
+            .flatMap(HttpClientRequest::rxSend)
+            .flatMapPublisher(response -> {
+                assertThat(response.statusCode()).isEqualTo(200);
+                return response.toFlowable();
+            })
+            .test()
+            .await()
+            .assertComplete()
+            .assertNoErrors();
+    }
+
+    @Test
+    @DeployApi(value = { "/apis/v2/with_absolute_url.json" })
+    public void should_route_to_url_when_request_match(HttpClient client) throws InterruptedException {
```

Comment:
As discussed, it could be nice to rely on a wiremock instance instead of a thirdparty website on which we do not have control. :)